src/components/plot/index.tsx

```diff
@@ -125,17 +131,35 @@ export class PlotBase extends React.PureComponent<PlotProps, PlotState> {
           onMouseEnter={this.onMouseEnter}
           onMouseLeave={this.onMouseLeave}
         >
-          <VegaLite spec={spec}/>
+          <VegaLite spec={spec} logger={this.plotLogger}/>
         </div>
         {notesDiv}
       </div>
     );
   }
 
+  protected componentDidMount() {
```

Comment:
Why do we need to flush this during "componentDidMount()"?  

Plus, I thought we agree that we can just create one log action per warning.  

That said, if there is a good solution, I am okay with flushing them all at once, but I'm not sure if this is the right trigger.  Isn't componentDidMount triggered only when it is mounted the first time?

Since it doesn't seem right this gonna be correct, why don't we just make logger trigger `handleAction`?